### DISCUSSION TITLE
Custom Main Menu Inquiry

### DISCUSSION THREAD

**Discussion by iton0** [AlmasB/FXGL#1376] [Q&A]
How would I customize the main menu such that I could have multiple buttons that start different FXGL games?

**Comment by AlmasB** [comment]
Hi, here's the link to [customise menus](https://github.com/AlmasB/FXGL/wiki/Customizing-Menus-%28FXGL-11%29) in general.

Depending on how you want to structure your games and complexity of games, there may be different options. It may be that you have 1 FXGL GameApplication that checks what button was pressed in the menu, then loads the appropriate game.

If games are quite complex, requiring their own menus, you can try separating the menu from FXGL and instead use [an embedded mode](https://github.com/AlmasB/FXGL/blob/10ee77c01791be7c9e6c81ed1ccf187b651e07b4/fxgl-samples/src/main/java/advanced/EmbeddedSample.java).

**Reply by iton0** [reply]
My structure right now is that I have a main FXGL GameApplication that acts as the Main Menu. However, I have other FXGL GameApplication classes that are the games themselves. Based on the Customizing Menus wiki, how would i add a button that calls one of these games (let's say the class name for a game is Game1)? Below is what I currently have, but I am unsure what the proper event handler is in this case:

```java 
Button game1 = FXGL.getUIFactoryService().newButton("Game 1");
game1.setOnMouseClicked(); // which event handler do I use to call the Game1 class
```

**Reply by AlmasB** [reply]
I see, I am not sure there is an easy way to do this, as FXGL won't allow multiple instances running simultaneously. 

However, if you were to change your Main Menu to just a normal JavaFX application, then you would simply call:

```
button.setOnMouseClicked(e -> GameApplication.embeddedLaunch(new MyGame1()));
```

as in the embedded sample above.
